### PR TITLE
Feature/project configurations

### PR DESCRIPTION
**Pull request by oleksandrchainsafe**

### CODE REVIEW

File: src/UnityPackage/Assets/Plugins/Web3/Scripts/ProjectConfigScriptableObject.cs [hunk 1]
```diff
@@ -35,9 +37,11 @@ public string Network
...
     }
 
-    public string RPC
+    public string Rpc
     {
         get => rpc;
         set => rpc = value;
     }
+
+    public static ProjectConfigScriptableObject LoadDefault() => Resources.Load<ProjectConfigScriptableObject>(DefaultAssetPath);
```
Comment (Arshia001) [inline src/UnityPackage/Assets/Plugins/Web3/Scripts/ProjectConfigScriptableObject.cs:46]: I have an entire static class dedicated to loading and saving the project config in the UPM PR. Let's put this PR on top of that and use that class.

File: src/UnityPackage/Assets/Web3Unity/Scripts/Scenes/WebLogin.cs [hunk 2]
```diff
@@ -22,12 +22,12 @@ public class WebLogin : MonoBehaviour
     void Start()
     {
         // loads the data saved from the editor config
-        projectConfigSO = (ProjectConfigScriptableObject)Resources.Load("ProjectConfigData", typeof(ScriptableObject));
-        PlayerPrefs.SetString("ProjectID", projectConfigSO.ProjectID);
-        PlayerPrefs.SetString("ChainID", projectConfigSO.ChainID);
+        projectConfigSO = ProjectConfigScriptableObject.LoadDefault();
+        PlayerPrefs.SetString("ProjectID", projectConfigSO.ProjectId);
```
Comment (Arshia001) [inline src/UnityPackage/Assets/Web3Unity/Scripts/Scenes/WebLogin.cs:26]: Do we even need the player prefs any more? I suppose we have to keep them so as to stay backwards-compatible, but we don't need them in the core of the SDK, only the compat layer.

**Review by Arshia001 — CHANGES_REQUESTED** [review]

File: src/UnityPackage/Assets/Web3Unity/Scripts/Scenes/WebLogin.cs (hunk 2 above)
Comment (sneakzttv) [reply src/UnityPackage/Assets/Web3Unity/Scripts/Scenes/WebLogin.cs:26]: we shouldn't need them there if they're being loaded in elsewhere and said new class is being referenced in the appropriate calls.

**Comment by sneakzttv** [comment]
linter has failed also

File: src/UnityPackage/Assets/Web3Unity/Scripts/Scenes/WebLogin.cs (hunk 2 above)
Comment (oleksandrchainsafe) [reply src/UnityPackage/Assets/Web3Unity/Scripts/Scenes/WebLogin.cs:26]: I didn't change anything here as it's out of the scope if this task

**Review by sneakzttv — DISMISSED** [review]

File: src/UnitySampleProject/Assets/Scripts/Scenes/WebLogin.cs [hunk 3]
```diff
@@ -22,9 +22,9 @@ public class WebLogin : MonoBehaviour
     void Start()
     {
         // loads the data saved from the editor config
-        projectConfigSO = (ProjectConfigScriptableObject)Resources.Load("ProjectConfigData", typeof(ScriptableObject));
-        PlayerPrefs.SetString("ProjectID", projectConfigSO.ProjectID);
-        PlayerPrefs.SetString("ChainID", projectConfigSO.ChainID);
+        projectConfigSO = ProjectConfigScriptableObject.LoadDefault();
```
Comment (Arshia001) [inline src/UnitySampleProject/Assets/Scripts/Scenes/WebLogin.cs:25]: This has to be `ProjectConfigUtilities.Load()`. I think you made the same mistake as me: not having WebGL as the active platform when working on the codebase.

**Pull request merged**